addons/account/models/mail_scheduled_message.py

```diff
@@ -0,0 +1,111 @@
+from odoo import api, fields, models, _
+from odoo.fields import Datetime
+
+
+class MailScheduledMessage(models.Model):
+    _inherit = 'mail.scheduled.message'
+
+    # JSON field to store all move related data to generate and send invoice
+    scheduled_send_payload = fields.Json()
+    from_move_send_wizard = fields.Boolean()
+
+    @api.ondelete(at_uninstall=False)
+    def _post_log_note_at_unlink(self):
+        for scheduled_message in self:
+            if scheduled_message.is_note and scheduled_message.from_move_send_wizard and not scheduled_message.env.context.get('from_post'):
+                move_id = self.env['account.move'].browse(scheduled_message.res_id)
+                user = self.env.user
+                create_date = Datetime.context_timestamp(user, Datetime.from_string(scheduled_message.create_date)).strftime('%Y-%m-%d %H:%M:%S')
+                now = Datetime.context_timestamp(user, Datetime.now()).strftime('%Y-%m-%d %H:%M:%S')
+                replace_body = _('scheduled by %(author)s at %(create_date)s is cancelled by %(current_user)s at %(now)s',
+                    author=scheduled_message.author_id.name,
+                    create_date=create_date,
+                    current_user=user.name,
+                    now=now,
+                )
+                body = scheduled_message.body.replace('is scheduled to send', replace_body)
+                move_id.message_post(
+                    author_id=scheduled_message.author_id.id,
+                    body=body,
+                    subtype_xmlid='mail.mt_note',
+                )
+
+    def post_message(self):
+        super(MailScheduledMessage, self.with_context(post_manual=True)).post_message()
```

Comment:
Avoid that kind of context key (post_manual, from_post), ... it makes code hard to read and understand, hides behavior change in context, ... globally a bad idea to have context-based code.